### PR TITLE
Add track and cluster weights decoders and their comparators

### PR DESCRIPTION
**Pull request by dudarboh**
Following change in `LCRelationNavigator` here: https://github.com/iLCSoft/LCIO/pull/163

Here are added functions to derive "Track" and "Cluster" weights from PFO-MCParticle LCRelation collection, which are encoded (and documented ONLY) by this processor: https://github.com/iLCSoft/MarlinReco/blob/0ed180e0ee3264f733bd86420debc6533b84e644/Analysis/RecoMCTruthLink/include/RecoMCTruthLinker.h#L27-L44

Also comparator functions are added to make more convenient use with the new LCRelationNavigator.

BEGINRELEASENOTES

- Add function to extract decoded Track and Cluster weights from encoded LCRelation collection for PFO/MCParticles.
- Add comparator functions for these weights encodings

ENDRELEASENOTES

### CODE REVIEW

File: source/include/MarlinUtil.h [hunk 1]
```diff
@@ -61,6 +61,44 @@ namespace MarlinUtil {
...
+   * set by the MarlinReco/Analysis/RecoMCTruthLink/include/RecoMCTruthLinker.h 
+   * for the ReconstructedParticle-MCParticle (or vise-versa) type of relations.
+   */
+  inline bool compareClusterWeights(float a, float b) {
+    return getClusterWeight(a) < getClusterWeight(b);
+  }
+
+
+  // ...
+
```
Comment (tmadlener) [inline source/include/MarlinUtil.h:101]:
```suggestion
```

File: source/include/MarlinUtil.h [hunk 2]
```diff
@@ -61,6 +61,44 @@ namespace MarlinUtil {
   double getEnergyDepositedInFullCalorimeter(lcio::LCEvent* evt);
 
 
+  /** Return track weight contribution encoded as trackwgt = (int(wgt)%10000)/1000. by
+   * the MarlinReco/Analysis/RecoMCTruthLink/include/RecoMCTruthLinker.h for the
+   * ReconstructedParticle-MCParticle (or vise-versa) type of relations.
+   * 
+   */
+  inline float getTrackWeight(float encodedWeight){
+      return ( int(encodedWeight) % 10000 ) / 1000.;
```
Comment (tmadlener) [inline source/include/MarlinUtil.h:70]:
```suggestion
      return float( int(encodedWeight) % 10000 ) / 1000.f;
```
Just to not trigger a potentialconversion warnings, make all of the explicit.

File: source/include/MarlinUtil.h [hunk 3]
```diff
@@ -61,6 +61,44 @@ namespace MarlinUtil {
...
+  }
+
+
+  /** Return cluster weight contribution encoded as clusterwgt = (int(wgt)/10000)/1000. by
+   * the MarlinReco/Analysis/RecoMCTruthLink/include/RecoMCTruthLinker.h for the
+   * ReconstructedParticle-MCParticle (or vise-versa) type of relations.
+   */
+  inline float getClusterWeight(float encodedWeight){
+      return ( int(encodedWeight) / 10000 ) / 1000.;
```
Comment (tmadlener) [inline source/include/MarlinUtil.h:79]:
```suggestion
      return float( int(encodedWeight) / 10000 ) / 1000.f;
```

**Review by tmadlener — COMMENTED** [review]
Looks good, I only have minor comments. Technically we could mark all of these as `constexpr`, but I am not sure they would ever be used in such a context.

**Comment by tmadlener** [comment]
I think it would actually make sense to also add the encoding functionality here, and then make the reco <-> mc truth linker use that. Doesn't change anything really, but it would give a few magic numbers a "name".

**Comment by dudarboh** [comment]
Done.
https://github.com/iLCSoft/MarlinReco/pull/108/files

**Review by tmadlener — APPROVED** [review]

File: source/include/MarlinUtil.h [hunk 4]
```diff
@@ -61,6 +61,32 @@ namespace MarlinUtil {
   double getEnergyDepositedInFullCalorimeter(lcio::LCEvent* evt);
 
 
+  /** Return track weight contribution encoded as trackwgt = (int(wgt)%10000)/1000. by
+   * the MarlinReco/Analysis/RecoMCTruthLink/include/RecoMCTruthLinker.h for the
```
Comment (tmadlener) [inline source/include/MarlinUtil.h:65]:
```suggestion
   * the encodeTrackAndClusterWeights function for the
```

File: source/include/MarlinUtil.h [hunk 5]
```diff
@@ -61,6 +61,32 @@ namespace MarlinUtil {
...
+   * ReconstructedParticle-MCParticle (or vise-versa) type of relations.
+   * 
+   */
+  inline float getTrackWeight(float encodedWeight){
+      return float( int(encodedWeight) % 10000 ) / 1000.f;
+  }
+
+
+  /** Return cluster weight contribution encoded as clusterwgt = (int(wgt)/10000)/1000. by
+   * the MarlinReco/Analysis/RecoMCTruthLink/include/RecoMCTruthLinker.h for the
```
Comment (tmadlener) [inline source/include/MarlinUtil.h:75]:
```suggestion
   * the encodeTrackAndClusterWeights function for the
```

**Pull request merged**